azurelinuxagent/ga/guestagent.py

```diff
@@ -16,6 +16,7 @@
 AGENT_ERROR_FILE = "error.json"  # File name for agent error record
 AGENT_MANIFEST_FILE = "HandlerManifest.json"
 MAX_FAILURE = 3  # Max failure allowed for agent before declare bad agent
+AGENT_UPDATE_COUNT_FILE = "update_attempt.json"  # File for tracking agent update attempt count
```

Comment:
we should also clean up this file during deprovisioning

Reply:
This file inside agent pkg, we delete Agent directory if I'm not mistaken

Reply:
Yes you're right. In temu issue customer had an image that was not deprovisioned correctly. When agent detects this state, it attempts to remove some stale files to minimize unexpected issues. For the TEMU PR, we added published_hostname to that list of files. I think we should add this too.

azurelinuxagent/pa/deprovision/default.py
We attempt to remove files in del_lib_dir_files, called by setup_changed_unique_id 
https://github.com/maddieford/WALinuxAgent/blob/0cd86173d40fc97a8eb21e199c2409d2d334fee4/azurelinuxagent/pa/deprovision/default.py#L154
https://github.com/maddieford/WALinuxAgent/blob/0cd86173d40fc97a8eb21e199c2409d2d334fee4/azurelinuxagent/pa/deprovision/default.py#L216